app/models/concerns/public_profile.rb

```diff
@@ -0,0 +1,35 @@
+module PublicProfile
+  extend ActiveSupport::Concern
+
+  def share_url(url)
+    update_public_profile_key if public_profile_key.blank?
+    form_url(url, public_profile_key)
+  end
+
+  private
+
+  def form_url(url, query_param)
+    parsed = URI.parse(url)
+    query = if parsed.query
+      CGI.parse(parsed.query)
+    else
+      {}
+    end
+
+    query["key"] = query_param
+
+    parsed.query = URI.encode_www_form(query)
+    parsed.to_s
+  end
+
+  def update_public_profile_key
+    update!(public_profile_key: generate_token)
+  end
+
+  def generate_token
+    loop do
+      public_key = SecureRandom.hex(10)
```

Comment:
```suggestion
      public_key = SecureRandom.hex(4)
```

I know it creates a higher chance of collision but then it is 8 characters like the developer profile slug.